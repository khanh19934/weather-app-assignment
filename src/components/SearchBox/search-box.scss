.search-box {
  position: relative;
  width: 40%;
  animation: scaleInput 0.3s;
  .form-control-feedback {
    position: absolute;
    z-index: 2;
    display: block;
    width: 1.375rem;
    height: 1.375rem;
    line-height: 2rem;
    text-align: center;
    pointer-events: none;
    color: #aaa;
    left: 0.65rem;
  }

  .close-search {
    position: absolute;
    z-index: 2;
    display: block;
    width: 1.375rem;
    height: 1.375rem;
    line-height: 2rem;
    text-align: center;
    cursor: pointer;
    color: #aaa;
    right: 0.65rem;
    top: 0;
  }

  .form-control {
    padding-left: 2.375rem;
  }

  &__suggestion {
    position: absolute;
    width: 100%;
    -webkit-box-shadow: 2px -1px 9px 0px rgba(0, 0, 0, 0.35);
    -moz-box-shadow: 2px -1px 9px 0px rgba(0, 0, 0, 0.35);
    box-shadow: 2px -1px 9px 0px rgba(0, 0, 0, 0.35);
    max-height: 300px;
    overflow: auto;
    border-bottom-left-radius: 7px;
    border-bottom-right-radius: 7px;
  }

  &__suggestion-item {
    display: flex;
    padding: 10px;
  }
}

@keyframes scaleInput {
  from {
    transform: scale(0);
  }
  to {
    transform: scale(1);
  }
}
