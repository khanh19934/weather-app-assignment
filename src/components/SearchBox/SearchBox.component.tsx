import React from 'react';
import { ReactComponent as SearchIcon } from '../../assets/icons/search_icon.svg';
import { ReactComponent as CloseIcon } from '../../assets/icons/close_icon.svg';
import './search-box.scss';

interface IProps {
  handleSearchLocation(e: React.ChangeEvent<HTMLInputElement>): void;
  handleToggleShowSearchBox(): void;
  handleSubmitSearch(e: React.KeyboardEvent<HTMLInputElement>): Promise<void>;
}

const SearchBox: React.FC<IProps> = ({
  handleSearchLocation,
  handleToggleShowSearchBox,
  handleSubmitSearch,
}: IProps) => {
  return (
    <div className='form-group search-box'>
      <div className='form-control-feedback'>
        <SearchIcon />
      </div>
      <input
        autoFocus
        onKeyDown={handleSubmitSearch}
        onChange={handleSearchLocation}
        type='text'
        className='form-control'
        placeholder='Search by city name...'
        data-testid='searchBox'
      />
      <div className='close-search' onClick={handleToggleShowSearchBox}>
        <CloseIcon />
      </div>
    </div>
  );
};

export default SearchBox;
