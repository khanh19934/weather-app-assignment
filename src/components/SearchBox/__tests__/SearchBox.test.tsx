import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from '../SearchBox.component';

describe('Snapshot test', () => {
  test('render correctly', () => {
    const tree = renderer
      .create(
        <SearchBox
          handleSearchLocation={jest.fn}
          handleSubmitSearch={jest.fn}
          handleToggleShowSearchBox={jest.fn}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
