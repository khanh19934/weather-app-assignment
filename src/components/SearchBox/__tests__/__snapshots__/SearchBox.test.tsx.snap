// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="form-group search-box"
>
  <div
    className="form-control-feedback"
  >
    <svg>
      search_icon.svg
    </svg>
  </div>
  <input
    autoFocus={true}
    className="form-control"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search by city name..."
    type="text"
  />
  <div
    className="close-search"
    onClick={[Function]}
  >
    <svg>
      close_icon.svg
    </svg>
  </div>
</div>
`;
